src/main/java/bobthebot/tasks/ToDoList.java

```diff
@@ -0,0 +1,236 @@
+
+                    String regex = "(\\d{4})-(\\d{2})-(\\d{2}) (\\d{2})(\\d{2})";
+                    // invalid deadline format
+                    if (!by.trim().matches(regex)) {
+                        String errorString = "\tInvalid formatting for deadline entered!\n";
+                        errorString += "\tWrite your deadlines in the following format: YYYY-MM-DD HHHH";
+                        Ui.printErrorMessage(errorString);
+                    } else if (by.trim().matches(regex) && !isAfter) {
+                        String errorString = "\tI might be a non-sentient robot but you seem to be a time traveller!\n";
+                        errorString += "\tPlease input deadlines BEFORE the current date and time.";
+                        Ui.printErrorMessage(errorString);
+                    } else {
+                        Task task = new Deadline(deadline[0], deadline[1]);
+                        list.add(task);
+                        storage.store(list);
+
+                        Ui.taskAddedMessage(task, this);
+                    }
+                }
+            } catch (ArrayIndexOutOfBoundsException e) {
+                String errorString = "\tInvalid formatting for deadline entered!\n";
+                errorString += "\tWrite your deadlines in the following format: \n";
+                errorString += "\t=> deadline <deadline-name> by <date>";
+                Ui.printErrorMessage(errorString);
+            } catch (DateTimeParseException e) {
+                String errorString = "\tInvalid formatting for deadline entered!\n";
+                errorString += "\tWrite your dates in the following format: YYYY-MM-DD HHHH";
+
+                Ui.printErrorMessage(errorString);
+            }
+        } else if (command.startsWith("event")) {
+            try {
+                command = command.replace("event ", "").trim();
+                if (command.length() == 0) {
+                    String errorString = "\tThe description of event cannot be empty!\n";
+                    errorString += "\tIt's impossible to go for something that does not exist...\n\n";
+                    errorString += "\tWrite your events in the following format: \n";
+                    errorString += "\t=> event <event-name> by <date>";
+
+                    Ui.printErrorMessage(errorString);
+                } else {
+                    String[] event = command.split(" /at ");
+
+                    String at = event[1];
+
+                    // generate boolean indicating if the deadline is before or after current date
+                    // and time
+                    LocalDateTime currDate = LocalDateTime.now();
+                    DateTimeFormatter format = DateTimeFormatter.ofPattern("uuuu-MM-dd kkmm");
+                    LocalDateTime deadlineDate = LocalDateTime.parse(at, format);
+                    Boolean isAfter = deadlineDate.isAfter(currDate);

```

Comment:
```suggestion
                    Boolean isDeadlineAfterCurrentDate = deadlineDate.isAfter(currDate);

```

Perhaps it might be better to have a more explicit variable naming